File: codeforces/practice/grade_allocation.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int T;
    cin >> T;
    for (int t = 0; t < T; t++) {
        long nStudents, maxScore;
        cin >> nStudents >> maxScore;
        long me;
        cin >> me;

        long total;
        for (int i = 1; i < nStudents; i++) {
            int score;
            cin >> score;
            total += score;
        }

        cout << min(total, maxScore) << endl;
    }
}

File: codeforces/practice/grade_allocation.py
T = int(input())
for t in range(T):
    students, maxScore = [int(x) for x in input().split(' ')]
    scores = [int(x) for x in input().split(' ')]
    me = scores[0]
    total = sum(scores[1:])
    print(min(total + me, maxScore))

